lib/AST/NameLookup.cpp

```diff
@@ -1253,38 +1253,15 @@ void ClassDecl::recordObjCMethod(AbstractFunctionDecl *method) {
   vec.push_back(method);
 }
 
-static bool checkAccessibility(const DeclContext *useDC,
-                               const DeclContext *sourceDC,
-                               Accessibility access) {
-  if (!useDC)
+bool ValueDecl::isAccessibleFrom(const DeclContext *DC) const {
+  auto access = getFormalAccess(DC);
+  if (!DC)
```

Comment:
This seems backwards to me. The only use for the formal-access-as-seen-from-DC at this point is when DC is null.